File: python/rotational-cipher/rotational_cipher.py
def rotate(text, key):
    return ''.join(list(map(lambda char: cipher(char, key), text)))

def cipher(char, key):
    if (char >= 'a' and char <= 'z'):#  or (char >= 'A' and char <= 'Z'):
        return chr((ord(char) - ord('a') + key) % 26 + ord('a'))
    elif char >= 'A' and char <= 'Z':
        return chr((ord(char) - ord('A') + key) % 26 + ord('A'))
    else:
        return char

File: haskell/rotational-cipher/src/RotationalCipher.hs
module RotationalCipher (rotate) where

import Data.Char (isAlpha, isAscii, ord, chr)

rotate :: Int -> String -> String
rotate n str = map (encrypt n) str

encrypt :: Int -> Char -> Char
encrypt n c
  | c >= 'a' && c <= 'z' = chr $ ord 'a' + ((ord c) - (ord 'a') + n) `mod` 26
  | c >= 'A' && c <= 'Z' = chr $ ord 'A' + ((ord c) - (ord 'A') + n) `mod` 26
  | otherwise = c
